job-service/src/main/java/com/hpe/caf/services/job/utilities/ExpirationPolicyHelper.java

```
@@ -0,0 +1,78 @@
+
+public final class ExpirationPolicyHelper
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ExpirationPolicyHelper.class);
+
+    private ExpirationPolicyHelper()
+    {
+    }
+
+    public static List<String> toDBString(final ExpirationPolicy expirationPolicy)
+    {
+        final List<String> policyList = new ArrayList<>();
+        final StringBuilder sb = new StringBuilder();
+        buildPolicy(policyList, sb, "Active,", expirationPolicy.getActive());
+        buildPolicy(policyList, sb, "Cancelled,", expirationPolicy.getCancelled());
+        buildPolicy(policyList, sb, "Completed,", expirationPolicy.getCompleted());
+        buildPolicy(policyList, sb, "Failed,", expirationPolicy.getFailed());
+        buildPolicy(policyList, sb, "Waiting,", expirationPolicy.getWaiting());
+        buildPolicy(policyList, sb, "Paused,", expirationPolicy.getPaused());
+        sb.setLength(0);
+        sb.append("(,,");
+        sb.append("Expired,")
+            .append(ExpirationOperationEnum.DELETE.toString()).append(",")
+            .append(expirationPolicy.getExpired().getExpiryTime())
+            .append(")");
+        policyList.add(sb.toString());
+
+        return policyList;
+    }
+
+    private static void buildPolicy(final List<String> policyList, final StringBuilder sb, final String status, final Policy policy)
+    {
+        validateExpiryTime(policy);
+        sb.setLength(0);
+        sb.append("(,,");
+        sb.append(status)
+            .append(policy.getOperation().toString()).append(",")
+            .append(policy.getExpiryTime())
+            .append(")");
+        policyList.add(sb.toString());
+    }
+
+    private static void validateExpiryTime(final Policy policy)
+    {
+        final String expiryTime = policy.getExpiryTime();
+        try {
+            DateHelper.validate(expiryTime);
+        } catch (final BadRequestException e) {
+            LOG.error("invalid expiry_time {}", expiryTime);
+        }
```

Is it correct that this is just logging here?  What is the intention?

---

We should rethrow the exception, done in the new code.
I remodelled so we check and pass the expiration policy earlier, reaching an exception earlier if any issue.
Also, moved the date validation from the helper onto the PolicyBuilder where it makes more sense I believe.